inc/audio_input.hpp

```diff
@@ -1,46 +1,99 @@
 //  For licensing see accompanying LICENSE file.
 //  Copyright © 2024 Argmax, Inc. All rights reserved.
-
 #pragma once
 
-#include <fstream>
 #include <iostream>
+#include <fstream>
 #include <string>
-#include <tflite_model.hpp>
 #include <vector>
 
-using namespace std;
+#define SDL_MAIN_HANDLED
+#include <SDL3/SDL.h>
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL_audio.h>
+
+#include <backend_class.hpp>
+
+#define SAMPLE_FREQ      16000 
+#define SAMPLE_FMT_DEF   0          // AV_SAMPLE_FMT_NONE in ffmpeg
+#define SAMPLE_FMT_FLT   3          // AV_SAMPLE_FMT_FLT in ffmpeg
+#define SAMPLE_FMT_S16P  6          // AV_SAMPLE_FMT_S16P in ffmpeg
+#define SAMPLE_FMT_FLTP  8          // AV_SAMPLE_FMT_FLTP in ffmpeg
+
+class AudioBuffer {
+private: 
+    SDL_AudioSpec* _source_spec;
+    SDL_AudioSpec* _target_spec;
+    SDL_AudioStream* _stream;
+    mutex _mutex;
+
+    // target buf associated, with 16khz, mono PCM data
+    uint32_t _end_index; // unit of short int
+    uint32_t _cap_bytes;
+    short int *_buffer;
+    int _bytes_per_sample;
 
-class AudioChunk {
-   public:
-    int start_index;
-    int num_samples;
-    float* data;
+public:
+    AudioBuffer();
+    ~AudioBuffer();
+
+    void initialize(
+        SDL_AudioSpec* src_spec,
+        SDL_AudioSpec* tgt_spec
+    );
+    void uninitialize();
+    bool empty_source();
+
+    int append(int bytes, char* buffer = nullptr);
+    int samples(int desired_samples = 0);
+    void consumed(int samples);
+    short int* get_buffer() { return _buffer; }
 };
 
-class AudioInputModel : public TFLiteModel {
-   public:
-    AudioInputModel(const string input_file);
+class AudioInputModel: public MODEL_SUPER_CLASS {
```

Comment:
let's make a doc regarding the camelcase / snake case / enums & constants / etc. and align it with WhisperKit style guidelines.  then we'll add the formatter & linter.

Reply:
Got it, so do you want to do it for the next release, or even for the current v0.1?

Reply:
not important for v0.1